addons/lunch/models/lunch.py

```diff
@@ -0,0 +1,315 @@
+# -*- coding: utf-8 -*-
+
+import datetime
+from openerp import models, fields, api, _
+
+class lunch_order(models.Model):
+    """
+    lunch order (contains one or more lunch order line(s))
+    """
+    _name = 'lunch.order'
+    _description = 'Lunch Order'
+    _order = 'date desc'
+
+    @api.multi
+    def name_get(self):
+        return [(order.id, "%s %s" % (_('Lunch Order'), order.id)) for order in self]
+
+    @api.multi
+    def _default_alerts_get(self):
+        """
+        get the alerts to display on the order form
+        """
+        alert_msg = [alert.get_alert_message()
+                     for alert in self.env['lunch.alert'].search([])
+                     if alert.get_alert_message()]
+        return '\n'.join(alert_msg)
+
+    @api.multi
+    def _compute_alerts_get(self):
+        """
+        get the alerts to display on the order form
+        """
+        for order in self:
+            if order.state == 'new':
+                order.alerts = self._default_alerts_get()
+
+    @api.multi
+    def _default_get_previous_order_ids(self):
+        prev_order = self.env['lunch.order.line']
+        prev_order_res = prev_order.search([('user_id', '=', self.env.uid)], limit=20 ,order='id desc')
+
+#        return prev_order_res.ids
+
+        # If we return return prev_order_res.ids, we will have duplicates.
```

Comment:
This is ugly: http://pix.toile-libre.org/upload/original/1425296033.png
This is beautiful: http://pix.toile-libre.org/upload/original/1425296082.png

Duplicates comes from the fact that we simply retrieve the last 20 orders, whatever the order is.